modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/CacheIndexImpl.java

```diff
@@ -134,4 +138,24 @@ public CacheIndexImpl(RelCollation collation, String name, Index idx, IgniteCach
         // Empty index find predicate.
         return new IndexConditions();
     }
+
+    /**
+     * @return Number of local records.
+     */
+    public long scanCount(ExecutionContext<?> ectx) {
+        IndexingQueryFilter filter = new IndexingQueryFilterImpl(tbl.descriptor().cacheContext().kernalContext(),
```

Comment:
Let's add the same precondition as for `scan()` method: `if (group.nodeIds().contains(localNodeId) && idx != null)`. In case  `ectx.target().partitions` is empty (none of partitions should be fetched from this node) filter will still return data from all primary partitions.

Reply:
Fixed.